File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/DocumentUnitSearchInput.java
package de.bund.digitalservice.ris.caselaw.domain;

import java.time.LocalDate;
import java.util.UUID;
import lombok.Builder;

@Builder
public record DocumentUnitSearchInput(
    UUID uuid,
    String documentNumberOrFileNumber,
    String courtType,
    String courtLocation,
    LocalDate decisionDate,
    LocalDate decisionDateEnd,
    DocumentationOffice documentationOffice,
    Status status,
    boolean myDocOfficeOnly) {}

File: frontend/src/domain/documentUnitSearchInput.ts
export default class DocumentUnitSearchInput {
  public id?: string
  uuid?: string
  documentNumberOrFileNumber?: string
  courtType?: string
  courtLocation?: string
  decisionDate?: string
  decisionDateEnd?: string
  publicationStatus?: string
  withError?: string
  myDocOfficeOnly?: string

  constructor(data: Partial<DocumentUnitSearchInput> = {}) {
    Object.assign(this, data)
  }

  static fields = [
    "documentNumberOrFileNumber",
    "courtType",
    "courtLocation",
    "decisionDate",
    "decisionDateEnd",
    "status",
  ] as const
}
